fastpath/fastpath/core.py

```diff
@@ -1353,29 +1353,43 @@ def score_signal(msm: dict) -> dict:
     try:
         # https://github.com/ooni/probe/issues/2344
         tv = g_or(msm, "test_version", "0.0.0")
-        if parse_version(tv) <= parse_version("0.2.3"):
+
+        msmt_start_time = msm.get("measurement_start_time")
+        if msmt_start_time is None:
+            scores["accuracy"] = 0.0
+            return scores
+        start_time = datetime.strptime(msmt_start_time, "%Y-%m-%d %H:%M:%S")
+
+        if (
+            parse_version(tv) <= parse_version("0.2.4")
+            and parse_version(tv) > parse_version("0.2.3")
+            and start_time >= datetime(2023, 11, 7)
+        ):
+            # See https://github.com/ooni/probe/issues/2636
+            # See https://github.com/ooni/probe-cli/pull/1421
+            scores["accuracy"] = 0.0
+            return scores
+
+        if parse_version(tv) <= parse_version("0.2.3") and start_time >= datetime(
+            2023, 11, 7
+        ):
```

Comment:
I see we're using the same `start_time` for both `if`s. I understand the `start_time` of 2023-11-07 comes from this issue https://github.com/ooni/probe/issues/2627. We noticed that we were testing too many endpoints on 2023-11-28, according to https://github.com/ooni/probe/issues/2636.

I think there are two options here:

1.

```suggestion
        if (
            parse_version(tv) <= parse_version("0.2.4")
            and start_time >= datetime(2023, 11, 7)
        ):
            # See https://github.com/ooni/probe/issues/2636
            # See https://github.com/ooni/probe-cli/pull/1421
```

2. Or using 2023-11-28 as the cutoff date for the fist check and dropping the `and parse_version(tv) > parse_version("0.2.3")` line.

The former suggestion coalesces two cases that are related together. The second one keeps them separate but applies the same kind of checking patterns to the code.

Reply:
As discussed on slack it's probably better to be explicit and not collapse the two cases into one, so that it's more clear they are addressing specific different issues affecting different measurements.